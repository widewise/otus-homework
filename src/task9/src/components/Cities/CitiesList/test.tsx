import React from "react";
import renderer from "react-test-renderer";
import { CitiesList } from "./index";
import { MemoryRouter } from "react-router";

describe(`${CitiesList.name}`, () => {
    //arrange
    const data = [
        {
            id: "1",
            name: "City",
            data: []
        }
    ];
    let actual: any;

    describe(`rerender ${CitiesList.name}`, () => {
        beforeEach(() => {
            //act
            actual = renderer.create(<MemoryRouter><CitiesList data={data} /></MemoryRouter>).toJSON();
        });

        it('should match snapshot', () => {
            //assert
            expect(actual).toMatchSnapshot();
        });
    });
});