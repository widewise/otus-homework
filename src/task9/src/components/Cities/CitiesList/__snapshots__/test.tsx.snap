// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesList rerender CitiesList should match snapshot 1`] = `
<ul>
  <li>
    <a
      href="/1"
      onClick={[Function]}
    >
      City
    </a>
  </li>
</ul>
`;
